geopmdpy/OPTIMIZER_README.md

```diff
@@ -0,0 +1,265 @@
+# Bayesian Optimizer for GEOPM Control Parameters
+
+The Bayesian optimizer is a tool that uses Bayesian optimization to find
+optimal control parameter settings for applications. It integrates with the
+ControlGrid functionality to define the parameter space and uses subprocess
+execution to evaluate different configurations.
+
+## Features
+
+- **Bayesian Optimization**: Uses scikit-optimize for intelligent parameter space exploration
+- **ControlGrid Integration**: Leverages existing grid functionality for parameter space definition
+- **Flexible Application Evaluation**: Supports any application that can be launched via subprocess
+- **Regex-based Metric Extraction**: Uses Python-style regular expressions to extract performance metrics
+- **Configurable Timeouts**: Customizable application execution timeouts
+- **Comprehensive Logging**: Tracks optimization progress and evaluation history
+- **Debug Output**: Optional application stdout logging for troubleshooting
+
+## Requirements
+
+- Python 3.7+
+- scikit-optimize (`pip install scikit-optimize`)
+- GEOPM Python package (geopmdpy)
+
+## Usage
+
+### Basic Command Line Usage
+
+```bash
+python -m geopmdpy.optimizer \
+    --cpu-frequency package \
+    --metric-regex "Performance: ([0-9.]+)" \
+    --n-calls 50 \
+    --output-file geopmwrite.conf \
+    -- ./run.sh
+```
+
+An example run script for optimizing perf-per-watt for the stream benchmark:
+
+```bash
+#!/bin/bash
+
+TMP=$(mktemp)
+echo "CPU_POWER board 0" | geopmsession -r- ./stream_c.exe > ${TMP}
+RATE=$(grep 'Triad:' ${TMP}  | awk '{print $2}')
+PWR=$(grep '    mean: ' ${TMP} | awk '{print $2}')
+python3 -c "print(f'Perf-per-watt: {$RATE / $PWR}')"
+rm ${TMP}
+```
+
+### Command Line Arguments
+
+#### Control Parameters
```

Comment:
Nice-to-have: Warning when dimensionality exceeds ~20 or so.  Thinking about CPU and CORE domain

Reply:
#3927